src/components/AuthForm.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { useEffect, useState } from "react";
+import { object, string } from "yup";
+import classes from "../styles/AuthForm.module.scss";
+import { FieldNames } from "../common/enums/FieldNames";
+import { ERROR_MESSAGES } from "../common/constants/constants";
+import { useAuthentication } from "../common/hooks/AuthHooks";
+import { useLocation, useNavigate } from "react-router-dom";
+import Button from "./Button";
+
+const PASSWORD_MIN_CHAR = 6;
+
+interface AuthData {
+  username?: string;
+  email: string;
+  password: string;
+  confirmPassword?: string;
+}
+
+type FormError = {
+  fieldName: keyof AuthData;
+  message: string;
+};
+
+const AuthForm: React.FC = () => {
+  const [authData, setAuthData] = useState<AuthData>({
+    email: "",
+    password: "",
+  });
+
+  const [errors, setErrors] = useState<FormError[]>([]);
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const [isSuccess, setIsSuccess] = useState(false);
+
+  const { signIn, signUp } = useAuthentication();
+
+  const location = useLocation();
+
+  let navigate = useNavigate();
+
+  const isSignIn = location.pathname === "/signin";
+
+  useEffect(() => {
+    setAuthData({
+      email: "",
+      password: "",
+      username: "",
+      confirmPassword: "",
+    });
+    setErrors([]);
+    setErrorMessage(null);
+  }, [isSignIn]);
```

Comment:
yes, this will work. And this is a react way to handle form data reset. but why not group this in a function `resetFormData` and call that function when user is signed in. Like a callback in signIn function. so you send `resetFormData` as a prop to signIn function, that will be called inside the function if it is successful. 
The reason to go this route is explicit data reset. You see what is happening, and not rely on a side effect.